src/octopeer-github/main/Status.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * The Status class handles the correct storage of the status.
+ * The status object is stored in the Chrome storage.
+ * There are 4 flags: 0 (error), 1 (running), 2 (off) and 3 (standby).
+ */
+
+enum StatusCode {ERROR, RUNNING, OFF, STANDBY}
+
+class Status {
+
+    /**
+     * Status.NAME: the internal names of the enum StatusCode.
+     */
+    public static get NAME() {
+        return [
+            "error",
+            "running",
+            "off",
+            "standby",
+        ];
+    }
+
+    /**
+     * Status.MESSAGE: the message strings of the enum StatusCode.
+     */
+    public static get MESSAGE() {
+        return [
+            "Octopeer has crashed!",
+            "Octopeer is running.",
+            "Octopeer is turned off.",
+            "Octopeer is standby.",
+        ];
+    }
+
+    /**
+     * Set the error status.
+     */
+    public error() {
+        this.set(StatusCode.ERROR);
+    }
+
+    /**
+     * Set the running status.
+     */
+    public running() {
+        this.set(StatusCode.RUNNING);
+    }
+
+    /**
+     * Set the off status.
+     */
+    public off() {
+        this.set(StatusCode.OFF);
+    }
+
+    /**
+     * Set the standby status.
+     */
+    public standby() {
+        this.set(StatusCode.STANDBY);
+    }
+
+    /**
+     * Set any status and inform the popup and the icon that they should be updated
+     * @param status
+     */
+    public set(status: StatusCode) {
+        chrome.storage.local.set({
+            status: status,
+        });
+
+        chrome.runtime.sendMessage({line: Status.MESSAGE[status], path: this.getIcon(status)});
+        chrome.browserAction.setIcon({path: this.getIcon(status, 19)});
+    }
+
+    /**
+     * @param status        the StatusCode to get the icon for.
```

Comment:
I know the function of getIcon is clear based on just the name, but please give a short description of what getIcon does. 
The rest of the JSDoc in the document is fine. :)